CI fails with test dependencies: torch.AcceleratorError: CUDA error: device-side assert

CI fails with test dependencies: https://github.com/huggingface/trl/actions/runs/20458276141/job/58785094629?pr=4700
> torch.AcceleratorError: CUDA error: device-side assert triggered

Stacktrace:
```python
  _ ERROR at teardown of TestGRPOTrainer.test_training_vlm[trl-internal-testing/tiny-LlavaNextForConditionalGeneration] _
  [gw0] linux -- Python 3.12.12 /__w/trl/trl/.venv/bin/python3
  
      @pytest.fixture(autouse=True)
      def cleanup_gpu():
          """
          Automatically cleanup GPU memory after each test.
      
          This fixture helps prevent CUDA out of memory errors when running tests in parallel with pytest-xdist by ensuring
          models and tensors are properly garbage collected and GPU memory caches are cleared between tests.
          """
          yield
          # Cleanup after test
          gc.collect()
          if torch.cuda.is_available():
  >           torch.cuda.empty_cache()
  
  tests/conftest.py:33: 
  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
  
      def empty_cache() -> None:
          r"""Release all unoccupied cached memory currently held by the caching
          allocator so that those can be used in other GPU application and visible in
          `nvidia-smi`.
      
          .. note::
              :func:`~torch.cuda.empty_cache` doesn't increase the amount of GPU
              memory available for PyTorch. However, it may help reduce fragmentation
              of GPU memory in certain cases. See :ref:`cuda-memory-management` for
              more details about GPU memory management.
          """
          if is_initialized():
  >           torch._C._cuda_emptyCache()
  E           torch.AcceleratorError: CUDA error: device-side assert triggered
  E           Search for `cudaErrorAssert' in https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__TYPES.html for more information.
  E           CUDA kernel errors might be asynchronously reported at some other API call, so the stacktrace below might be incorrect.
  E           For debugging consider passing CUDA_LAUNCH_BLOCKING=1
  E           Compile with `TORCH_USE_CUDA_DSA` to enable device-side assertions.
  
  .venv/lib/python3.12/site-packages/torch/cuda/memory.py:224: AcceleratorError
```

--- a/scripts/generate_tiny_models.py
+++ b/scripts/generate_tiny_models.py
@@ -320,7 +320,8 @@ for model_id, model_class, dtype in [
     ("HuggingFaceM4/Idefics3-8B-Llama3", Idefics3ForConditionalGeneration, torch.bfloat16),
     ("HuggingFaceTB/SmolVLM2-2.2B-Instruct", SmolVLMForConditionalGeneration, torch.float32),
     ("llava-hf/llava-1.5-7b-hf", LlavaForConditionalGeneration, torch.float16),
-    ("llava-hf/llava-v1.6-mistral-7b-hf", LlavaNextForConditionalGeneration, torch.float16),
+    # Original model dtype is float16, but it triggers CUDA device side assert error (see GH-4741):
+    ("llava-hf/llava-v1.6-mistral-7b-hf", LlavaNextForConditionalGeneration, torch.bfloat16),
     ("OpenGVLab/InternVL3-8B-hf", InternVLForConditionalGeneration, torch.bfloat16),
     ("Qwen/Qwen2-VL-2B-Instruct", Qwen2VLForConditionalGeneration, torch.bfloat16),
     ("Qwen/Qwen2.5-VL-3B-Instruct", Qwen2_5_VLForConditionalGeneration, torch.bfloat16),

